### PR TITLE
fix: isolate expo-task-manager/expo-background-task to /wallet/expo/background

### LINKED ISSUE #486: Lazy `require()` of `expo-task-manager` / `expo-background-task` is invisible to Metro
## Issue

Importing from `@arkade-os/sdk/wallet/expo` and calling `defineExpoBackgroundTask` (or `registerExpoBackgroundTask` / `unregisterExpoBackgroundTask`) throws

> `expo-task-manager is required for background tasks. Install it with: npx expo install expo-task-manager`

at runtime on Expo / React Native, even with both `expo-task-manager` and `expo-background-task` installed.

This is the twin of arkade-os/boltz-swap#136, fixed by arkade-os/boltz-swap#146. The broken pattern was copied (or independently re-created) here in `src/wallet/expo/background.ts`.

## Cause

`src/wallet/expo/background.ts:39-59` loads both packages via lazy `require()` inside `requireTaskManager` / `requireBackgroundTask`:

```ts
function requireTaskManager(): TaskManagerModule {
    try {
        return require("expo-task-manager") as TaskManagerModule;
    } catch { … }
}
function requireBackgroundTask(): BackgroundTaskModule {
    try {
        return require("expo-background-task") as BackgroundTaskModule;
    } catch { … }
}
```

The package is built as ESM (`"type": "module"`, the `import` export condition points at `dist/esm/...`). tsc/Rollup rewrites bare `require` in ESM output to an aliased helper (`__require` / `createRequire(...)`). Metro's static dependency collector only recognises literal `require("…")` calls — the aliased form is invisible, so **`expo-task-manager` and `expo-background-task` never enter the bundle graph**. Metro dev's name-based fallback has nothing to resolve; production builds only accept numeric module IDs anyway. The CJS output uses native `require()`, but Metro picks the ESM variant via the `import` export condition, so consumers always hit the broken path.

Both packages are declared as optional peer-deps (`package.json` `peerDependencies` + `peerDependenciesMeta`), so they are genuinely soft-optional — the same constraint that motivated the split in `boltz-swap`.

## Additional symptom: silent failure in `ExpoWallet`

`ExpoWallet.setup()` (`src/wallet/expo/wallet.ts:187-199`) and `dispose()` (`:249-256`) wrap the dynamic `import("./background")` in a `try/catch` that swallows all errors with a comment claiming `expo-background-task not installed`. With this bug, that catch fires even when the package *is* installed, so OS-task registration silently does nothing and `ExpoWallet` looks like it works while running foreground-only.

## Validation: is a split necessary?

`src/wallet/expo/index.ts:3-7` re-exports the background functions directly from `./background`:

```ts
export {
    defineExpoBackgroundTask,
    registerExpoBackgroundTask,
    unregisterExpoBackgroundTask,
} from "./background";
```

So any consumer pulling `ExpoWallet` for foreground use already drags `background.ts` into the static graph. Just swapping the lazy `require()`s for static imports inside `background.ts` would regress non-Expo / non-mobile consumers of `/wallet/expo`:

- **react-native-web / web**: `expo-task-manager`'s `expo-module.config.json` declares `platforms: ["apple", "android"]` only — no web entry. Static imports in `/wallet/expo` would break web builds for any consumer who only wants `ExpoWallet`.
- **bare React Native without background tasks**: forces two unnecessary installs.
- **Node tooling / tests** reaching into `/wallet/expo`: same.

Splitting makes the requirement opt-in: importing `/wallet/expo/background` is the explicit signal "I'm on Expo Android/iOS and want the OS-scheduled task." `/wallet/expo` stays soft-dep-free, matching the current `package.json` intent.

## Recommended port of the fix

Mirroring arkade-os/boltz-swap#146:

1. **New subpath `@arkade-os/sdk/wallet/expo/background`** with its own `exports` entry in `package.json`. The new entry replaces the lazy `require()`s with static imports:
   ```ts
   import * as TaskManager from \"expo-task-manager\";
   import * as BackgroundTask from \"expo-background-task\";
   ```

2. **Remove the background re-exports from `src/wallet/expo/index.ts`.** Direct callers migrate to:
   ```ts
   import { defineExpoBackgroundTask } from \"@arkade-os/sdk/wallet/expo/background\";
   import { ExpoWallet } from \"@arkade-os/sdk/wallet/expo\";
   ```
   Breaking change — call it out in `CHANGELOG.md` under **Breaking Changes** with the migration snippet inline (per `FOUNDATION.md`).

3. **Drop OS-task registration from `ExpoWallet.setup()` / `dispose()`.** Remove the dynamic `import(\"./background\")` calls and the `try/catch` that hides the real error. Consumers call `registerExpoBackgroundTask` / `unregisterExpoBackgroundTask` explicitly from `/wallet/expo/background`. Also drop `ExpoBackgroundConfig.minimumBackgroundInterval` (and likely `taskName`) since OS registration moves out — keep `taskQueue`, `processors`, `foregroundIntervalMs`.

4. **Smoke test in CI**: after build, grep `dist/esm/wallet/expo/index.js` for `expo-task-manager` / `expo-background-task` and fail if found. Catches accidental re-exports.

5. **README + CHANGELOG migration callout** with the new import paths and the explicit `registerExpoBackgroundTask` call.

## References

- Twin issue: arkade-os/boltz-swap#136
- Fix PR (to port): arkade-os/boltz-swap#146

### PR DESCRIPTION
**Pull request by pietro909**
- Fix #486: `defineExpoBackgroundTask` (and `registerExpoBackgroundTask` / `unregisterExpoBackgroundTask`) failed at JS startup because lazy `require()` in `/wallet/expo` was invisible to Metro's static dependency collector, so `expo-task-manager` / `expo-background-task` never entered the bundle graph.
- Split task helpers into a new `@arkade-os/sdk/wallet/expo/background` subpath that uses **static** imports. `/wallet/expo` itself no longer references the optional Expo packages — react-native-web and Node consumers are unaffected.
- Drop OS-task registration from `ExpoWallet.setup()` / `.dispose()`. Consumers call `registerExpoBackgroundTask` / `unregisterExpoBackgroundTask` explicitly.
- README migration callout + teardown example updated.

## Testing

Direct port of arkade-os/boltz-swap#146, whose validated build pattern is in production in Trixie Wallet. Test changes:

- `test/wallet/expo/background.test.ts` rewritten to mock `expo-task-manager` / `expo-background-task` via `vi.hoisted` + `vi.mock` — the previous `Module.prototype.require` patching does not intercept static imports.
- `test/wallet/expo/wallet.test.ts` drops the `taskName` / `minimumBackgroundInterval` mocks (those fields are gone) and adds coverage for the new `warnOnRemovedBackgroundFields` JS-caller guard.
- Smoke-check after `pnpm build`: grep'ing `dist/{esm,cjs}/wallet/expo/{index,wallet}.js` shows zero runtime imports of `expo-task-manager` / `expo-background-task` — only `/wallet/expo/background.{js,cjs}` pulls them in.

## Summary by CodeRabbit

* **New Features**
  * Background helpers moved to a dedicated subpath and published so consumers can import OS registration helpers directly.
  * Added ambient TypeScript types for Expo background-task APIs.

* **Documentation**
  * Updated Expo background setup, examples, and architecture notes to show explicit register/unregister calls and new import paths.
  * Wallet setup examples updated to reflect removed config fields.

* **Breaking / API Changes**
  * Wallet no longer auto-registers OS background tasks; background config no longer accepts taskName or minimumBackgroundInterval.
  * Added runtime warning for removed legacy background fields.

* **Tests**
  * Background-task tests refactored to use static mocks; added tests for the removal-warning helper.

[Review Change Stack](https://app.coderabbit.ai/change-stack/arkade-os/ts-sdk/pull/487?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

**Review by ghost — APPROVED** [review]
## Review: arkade-os/ts-sdk#487 — isolate expo-task-manager/expo-background-task

**Verdict: APPROVE** ✅

Clean, well-motivated fix. The architectural split is sound and the implementation is tight.

### What this PR does

Fixes #486 by moving `expo-task-manager` / `expo-background-task` from lazy `require()` (invisible to Metro's static dep collector) into a dedicated `@arkade-os/sdk/wallet/expo/background` subpath with **static** imports. The `/wallet/expo` entrypoint no longer touches those packages at all, so react-native-web and Node consumers aren't affected.

Breaking change: `taskName` and `minimumBackgroundInterval` removed from `ExpoBackgroundConfig`; OS-level registration is now the consumer's explicit responsibility.

### Reviewed files

| File | Assessment |
|------|-----------|
| `package.json` | ✅ New subpath export correctly wired (types, ESM, CJS, default) |
| `src/wallet/expo/background.ts` | ✅ Static imports replace lazy require(); inline type declarations removed in favor of ambient `.d.ts` |
| `src/wallet/expo/expo-modules.d.ts` | ✅ Ambient declarations cover the subset of APIs used; sits inside `src/` so `tsconfig.json` `"include": ["src"]` picks it up |
| `src/wallet/expo/index.ts` | ✅ Background re-exports removed — consumers must import from the new subpath |
| `src/wallet/expo/wallet.ts` | ✅ `import type` from `./background` — erased at compile time, no runtime dep chain. `warnOnRemovedBackgroundFields` catches JS callers still passing stale fields |
| `src/worker/expo/README.md` | ✅ Migration table, updated examples, teardown example added |
| `test/wallet/expo/background.test.ts` | ✅ `vi.hoisted` + `vi.mock` replaces brittle `Module.prototype.require` patching — correct for static imports |
| `test/wallet/expo/wallet.test.ts` | ✅ Background mock removed (no longer imported by `wallet.ts`), `warnOnRemovedBackgroundFields` thoroughly tested including edge cases |

### Key architectural check

The critical invariant — `/wallet/expo` must not transitively import `expo-task-manager` or `expo-background-task` at runtime — is preserved:
- `wallet.ts:34` uses `import type { PersistedBackgroundConfig } from "./background"` → erased by TypeScript.
- `index.ts` re-exports only from `./wallet`, not from `./background`.
- The PR description confirms a post-build grep of `dist/` validates this.

### Minor observations (non-blocking)

1. **`DefineBackgroundTaskOptions` / `PersistedBackgroundConfig` no longer re-exported from `/wallet/expo`** — they were previously in `index.ts`. They're now only accessible via `@arkade-os/sdk/wallet/expo/background`. This is correct behavior (consumers of background types should import from the background subpath), but worth noting in changelog.

2. **`registerExpoBackgroundTask` default interval math** (`background.ts:162`): `(options?.minimumInterval ?? 15) * 60` converts minutes to seconds. The `expo-background-task` API expects seconds, so this is correct. The JSDoc correctly documents the param as minutes.

3. **No cross-repo breakage**: Searched all repos under `/root/arkana/repos/` — zero consumers currently import from `@arkade-os/sdk/wallet/expo`. The primary downstream consumer (Trixie Wallet) is referenced in the PR description as already validated against the equivalent boltz-swap#146 change.

### Protocol safety

This PR does **not** touch VTXO handling, transaction signing, forfeit paths, round lifecycle, connector trees, or exit paths. The background task handler reconstructs providers and calls `runTasks` + `extendVtxo` — but that logic is unchanged. The only changes are import/export topology and the consumer-facing registration API.

No protocol-critical review needed.

---

🤖 Reviewed by Arkana (opus)

File: src/wallet/expo/expo-modules.d.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+declare module "expo-task-manager" {
+    export function defineTask(
+        taskName: string,
+        executor: (body: {
+            data: unknown;
+            error: { code: string | number; message: string } | null;
+            executionInfo: { eventId: string; taskName: string };
+        }) => Promise<unknown>
+    ): void;
+}
```
[the remark this answers is not in the document]
Comment (pietro909) [reply src/wallet/expo/expo-modules.d.ts:24]: The executor we pass doesn't read data, error, or `executionInfo` at all. Current `Promise<unknown>` accepts our numeric return.  Adding the generic, `TaskManagerError` interface, and `appState` is type-accuracy work on members nobody uses, which directly conflicts with the file's documented "subset of APIs we actually use" purpose.

File: src/wallet/expo/expo-modules.d.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+declare module "expo-background-task" {
+    export const BackgroundTaskResult: { Success: 1; Failed: 2 };
+    export function registerTaskAsync(
+        taskName: string,
+        options?: { minimumInterval?: number }
+    ): Promise<void>;
+    export function unregisterTaskAsync(taskName: string): Promise<void>;
+}
```
[the remark this answers is not in the document]
Comment (pietro909) [reply src/wallet/expo/expo-modules.d.ts:33]: We don't call getStatusAsync or triggerTaskWorkerForTestingAsync and don't reference  `backgroundTaskStatus`. Adding them contradicts the file's subset-only purpose. The const `{ Success: 1; Failed: 2 }` shape is structurally identical to the `enum` for property access.
